### PR TITLE
Added devise jwt controllers

### PR DESCRIPTION
**Pull request by icoluccio**
## Summary
Changed default sign in gem for devise-jwt-controllers

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -11,8 +11,7 @@ gem 'puma', '~> 4.2.1'
 gem 'therubyracer', platforms: :ruby
 
 # Authentication
-gem 'devise_token_auth', '~> 1.0'
-gem 'omniauth'
+gem 'devise_jwt_controllers', github:'Wolox/devise_jwt_controllers', branch: 'release-0.1.3'
```
Comment (lucasVoboril) [inline Gemfile]: Why we don't publish this gem?

Comment (icoluccio) [reply Gemfile]: 👀

File: spec/support/shared_examples/with_authenticated_user.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+shared_context 'with authenticated user' do
+  let(:current_user) { create(:user, confirmed_at: Time.zone.now) }
+
+  before do
+    headers = { 'Accept' => 'application/json', 'Content-Type' => 'application/json' }
+    auth_headers = Devise::JWT::TestHelpers.auth_headers(headers, current_user)
+    request.headers.merge!(auth_headers)
+  end
+end
```
[the remark this answers is not in the document]
Comment (icoluccio) [reply spec/support/shared_examples/with_authenticated_user.rb]: Do we want to add those gems though? I should learn more about them before replying to this though

[the remark this answers is not in the document]
Comment (icoluccio) [reply spec/support/shared_examples/with_authenticated_user.rb]: Also not a huge fan of before(:all) for non-intensive stuff. It makes tests non-independent

[the remark this answers is not in the document]
Comment (lucasVoboril) [reply spec/support/shared_examples/with_authenticated_user.rb]: :raising_hand_man: My bad, sorry. I use this gem, and I've supposed it was in gemfile :sweat_smile:

[the remark this answers is not in the document]
Comment (icoluccio) [reply spec/support/shared_examples/with_authenticated_user.rb]: We can totally debate and use it though! i'm liking what I'm reading

[the remark this answers is not in the document]
Comment (holywyvern) [reply spec/support/shared_examples/with_authenticated_user.rb]: I don't really like to share the current_user in multiple test, is hard to maintain if you for example, create items for a current_user, and also it's not a big performance gain.

In projects where this is over used, the test are *really hard to read* and I believe test in rspec are meant to be **easy to read first**

File: Gemfile (hunk 1 above)
Comment (holywyvern) [reply Gemfile]: [*it **is** published*](http://rubygems.org/gems/devise_jwt_controllers). 🙄

**Review by lucasVoboril — DISMISSED** [review]
Only one suggestion to tag gem version :+1: nice

**Pull request closed without merging**